server/src/main/java/io/unitycatalog/server/persist/Repositories.java

```diff
@@ -32,7 +32,7 @@ public Repositories(SessionFactory sessionFactory, ServerProperties serverProper
 
     this.catalogRepository = new CatalogRepository(this, sessionFactory);
     this.schemaRepository = new SchemaRepository(this, sessionFactory);
-    this.tableRepository = new TableRepository(this, sessionFactory);
+    this.tableRepository = new TableRepository(this, sessionFactory, serverProperties);
```

Comment:
Passing the whole `serverProperties` to the `TableRepository` seems strange to me, because the database layer should not be aware of the whole server-side properties.  A better option is:  use passing the `managedTableEnabled`  boolean switch to the database layer.

Reply:
i am okay with the server properties being passed as a whole. today there is one property that affects the behavior. tomorrow there may be many more properties. we dont want to keep changing the constructor signature every time. So better in the long run to send the whole serverproperties in the repo

think of it like hadoopConf. hadoopConf passed round everywhere for exactly this reason.

Reply:
Okay, then let's keep it.